Character.cpp

```diff
@@ -98,20 +97,33 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        //assert(false);
+        levelUp(hitPoints,initialHitPoints);
+        levelUp(armor,initialArmorLevel);
+        levelUp(attackDamage,initialAttackDamage);
+
         std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
     }
 }
 
+void Character::levelUp(int& stat , std::unique_ptr<int>& initialStat )
+{
+    if(stat < *initialStat)
+        stat = *initialStat;
+    stat *= 1.1;
+    *initialStat = stat;
+}
```

Comment:
skip the `std::unique<>&` arg, and pass a reference to the `int` that it owns.
that will also save you from dereferencing it 3 times.
```
void levelUp(int& stat, int& initialStat)
```
- [ ] revise